### PR TITLE
Cart page updates

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Fix style and layout issues

Issue Number: N/A

## What is the new behavior?
Looks better, less code.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<div class="cart-wrapper">
+  <div class="cart-wrapper__wrapper">
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.html]: I think this should be named this `cart-wrapper__main` or something. Goes better with `cart-wrapper__side-pane`

File: apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.spec.ts [hunk 2]
```diff
@@ -58,10 +58,10 @@ class ProceedToCheckoutMock {}
 })
 class ContinueShoppingMock {}
 
-describe('TestCartAsyncWrapper', () => {
-  let component: TestCartAsyncWrapper;
-  let fixture: ComponentFixture<TestCartAsyncWrapper>;
-  let cartAsyncWrapperComponent: CartAsyncWrapperComponent;
+describe('TestCartWrapper', () => {
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.spec.ts]: This should be CartWrapper. Just the describe part, here. I must've messed this up when I wrote it originally.

File: apps/foundation-demo/src/app/cart/components/cart/cart.component.scss [hunk 3]
```diff
@@ -29,8 +20,13 @@
     font-size: $small-font-size;
 
     @include breakpoint(large) {
-      margin-left: 25px;
+      margin-left: 15px;
       float: none;
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/cart/components/cart/cart.component.scss]: Let's get rid of this float too. Should be able to with `grid` and `justify-self: right`

File: apps/foundation-demo/src/app/cart/components/cart/cart.component.html [hunk 4]
```diff
@@ -1,10 +1,12 @@
 <div class="cart">
-  <div class="cart__title" *ngIf="subtitle">
-    <span class="cart__subtitle">{{subtitle}}</span>
+  <div class="cart__header" *ngIf="title">
```
Comment (damienwebdev) [inline apps/foundation-demo/src/app/cart/components/cart/cart.component.html]: Why can't we use transclusion here? I feel like `title` is an unnecessary `Input`.

File: apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    <cart [cart]="cart" [title]="'Your Cart'"></cart>
+    <continue-shopping class="cart-wrapper__continue-shopping" *ngIf="isCartEmpty"></continue-shopping>
```
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.html]: I removed it on purpose. Didn't feel like it was needed anymore.

File: apps/foundation-demo/src/app/cart/components/cart/cart.component.html (hunk 4 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/cart/cart.component.html]: @lderrickable

Comment (lderrickable) [reply apps/foundation-demo/src/app/cart/components/cart/cart.component.html]: I think we can. I'll do a commit for it

Comment (lderrickable) [reply apps/foundation-demo/src/app/cart/components/cart/cart.component.html]: So I missed the ngIf before. I think that's needed for instances of cart without a header, which I think happens in checkout somewhere. Is there a way to do that with transclusion? I don't think there is. @damienwebdev

File: apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (lderrickable) [reply apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.html]: ok

File: apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.html (hunk 1 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.html]: done

File: apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.spec.ts (hunk 2 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.spec.ts]: done

File: apps/foundation-demo/src/app/cart/components/cart/cart.component.scss (hunk 3 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/cart/cart.component.scss]: fixed with `display: flex`

File: apps/foundation-demo/src/app/cart/components/cart-totals-item/cart-totals-item.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    <ng-container cart-total-value>{{value}}</ng-container>
+  </cart-totals-item>
+`})
+class TestCartTotalsItemWrapper {
+  label: string;
+  value: string;
+  emphasize: boolean;
+}
+
+fdescribe('CartTotalsItemComponent', () => {
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/cart/components/cart-totals-item/cart-totals-item.component.spec.ts]: fdescribe needs replaced with describe

File: apps/foundation-demo/src/app/cart/components/cart-totals-item/cart-totals-item.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+  it('can transclude a value', () => {
+    component.value="value";
+    fixture.detectChanges();
+
+    let cartTotalsItemComponent = fixture.debugElement.query(By.css('.cart-totals-item__value'));
+    expect(cartTotalsItemComponent.nativeElement.innerHTML).toContain('value');
+  });
+
+  fit('should add the emphasize class to cart-totals-item when emphasize is true', () => {
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/cart/components/cart-totals-item/cart-totals-item.component.spec.ts]: fit needs replaced with it

File: apps/foundation-demo/src/app/cart/components/continue-shopping/continue-shopping.directive.spec.ts [hunk 8]
```diff
@@ -34,7 +34,7 @@ describe('ContinueShoppingDirective', () => {
     expect(component).toBeTruthy();
   });
 
-  describe('when button is clicked', () => {
+  fdescribe('when button is clicked', () => {
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/cart/components/continue-shopping/continue-shopping.directive.spec.ts]: fdescribe

File: apps/foundation-demo/src/app/cart/components/proceed-to-checkout/proceed-to-checkout.component.spec.ts [hunk 9]
```diff
@@ -38,7 +43,7 @@ describe('ProceedToCheckoutComponent', () => {
     expect(fixture.debugElement.query(By.css('.button'))).toBeDefined();
   });
 
-  describe('when button is clicked', () => {
+  fdescribe('when button is clicked', () => {
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/cart/components/proceed-to-checkout/proceed-to-checkout.component.spec.ts]: fdescribe

**Comment by lderrickable** [comment]
I'm requesting that we don't add new features or refactors to a PR after requesting a review. Ideally, any commits made after the initial review should be only to address request changes. It makes reviewing easier.

File: apps/foundation-demo/src/app/cart/components/cart-totals-item/cart-totals-item.component.spec.ts (hunk 6 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/cart-totals-item/cart-totals-item.component.spec.ts]: fixed

File: apps/foundation-demo/src/app/cart/components/cart-totals-item/cart-totals-item.component.spec.ts (hunk 7 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/cart-totals-item/cart-totals-item.component.spec.ts]: fixed

File: apps/foundation-demo/src/app/cart/components/continue-shopping/continue-shopping.directive.spec.ts (hunk 8 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/continue-shopping/continue-shopping.directive.spec.ts]: fixed

File: apps/foundation-demo/src/app/cart/components/proceed-to-checkout/proceed-to-checkout.component.spec.ts (hunk 9 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/proceed-to-checkout/proceed-to-checkout.component.spec.ts]: fixed

**Comment by xelaint** [comment]
> I'm requesting that we don't add new features or refactors to a PR after requesting a review. Ideally, any commits made after the initial review should be only to address request changes. It makes reviewing easier.

@lderrickable yeah sorry about that. Will make sure to only request review when i'm completely done for future PRs.

File: apps/foundation-demo/src/app/cart/components/cart/cart.component.html (hunk 4 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/cart/components/cart/cart.component.html]: Why do we have to transclude or put it as an input if the text for the cart title is always going to be `Your Cart`

**Comment by xelaint** [comment]
@lderrickable build still failed

**Comment by lderrickable** [comment]
Ok, so the tests should hopefully pass now. Just a heads up though, that the tests that I "fixed" are pretty bad now. The problem is that you can't test an object for its type, if that type in coming from an interface. 
```
result: Cart = cartFactory.create();
expect(result instanceof Cart).toBeTruthy();
expect(result).toEqual(jasmine.any(Cart));
```
Neither of those tests work, because `instanceof` requires the right side to be a concrete class, while Cart is an interface.

**Comment by xelaint** [comment]
@lderrickable @damienwebdev rebased and fixed build errors from @daffodil/design import.

**Comment by xelaint** [comment]
@lderrickable fixed

**Comment by xelaint** [comment]
@lderrickable @damienwebdev need to be re-approved since i had to fix conflicts

**Pull request merged**